### PR TITLE
Proposal for Keypath

### PR DESCRIPTION
**Pull request by michael-lehew**

### CODE REVIEW

File: proposals/XXXX-key-paths.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	var friends: [Person]
+	var bestFriend: Person?
+}
+
+var han = Person(name: "Han Solo", friends: [])
+var luke = Person(name: "Luke Skywalker", friends: [han])
+
+let firstFriendsNameKeyPath = Person.friends[0].name
+
+let firstFriend = luke[path] // han
```
Comment (jjmanton) [inline proposals/XXXX-key-paths.md]: Should this be `let firstFriend = luke[firstFriendsNameKeyPath]` ?

Comment (michael-lehew) [inline proposals/XXXX-key-paths.md]: Yes it should.  That was a Last Minute Change™ where the find-replace failed. :(

**Comment by jido** [comment]
You have this:
> // or equivalently, with type inferred from context
> luke[keyPath: #keyPath(.friends[0].name)]
Does this not conflict with backward compatibility?
> There is no change or interaction with the #keyPath() syntax introduced in Swift 3. #keyPath(Person.bestFriend.name) will still produce a String

**Comment by michael-lehew** [comment]
This, along with other spelling matters regarding of this is being discussed on the swift-evolution list. 

-Michael

**Pull request merged**